.chloggen/component-refactor-datatype-6.yaml

```diff
@@ -0,0 +1,25 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: deprecation
+
+# The name of the component, or a single word describing the area of concern, (e.g. otlpreceiver)
+component: connector
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Deprecates `TracesRouterAndConsumer`, `NewTracesRouter`, `MetricsRouterAndConsumer`, `NewMetricsRouter`, `LogsRouterAndConsumer`, and `NewLogsRouter`. Use `TracesRouterAndConsumerWithPipelineIDs`, `NewTracesRouterWithPipelineIDs`, `MetricsRouterAndConsumerWithPipelineIDs`, `NewMetricsRouterWithPipelineIDs`, `LogsRouterAndConsumerWithPipelineIDs`, and `NewLogsRouterWithPipelineIDs` instead.
```

Comment:
I would be fine with breaking this as well in 1-step

Reply:
I will update the connector stuff to be breaking.

Reply:
Updated